src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql

```diff
@@ -23,10 +23,20 @@ RETURNS boolean AS $$
         FROM get_all_active_transactions() WHERE process_id = pBlockedPid;
     END IF;
 
-    RETURN EXISTS (
-      SELECT 1 FROM citus_internal_global_blocked_processes()
-        WHERE waiting_global_pid = mBlockedGlobalPid
-    );
+    IF current_setting('citus.isolation_test_check_all_blocks') = 'off' THEN
+      RETURN EXISTS (
+        SELECT 1 FROM citus_internal_global_blocked_processes()
+          WHERE waiting_global_pid = mBlockedGlobalPid
+          AND blocking_global_pid in (
+              SELECT citus_calculate_gpid(citus_backend_nodeid(), pid) FROM unnest(pInterestingPids) pid
```

Comment:
Hmm, so we only use the current node that is running the query (e.g., coordinator) `citus_backend_nodeid()`. 

Shouldn't we need to get `citus_backend_nodeid` from the workers where `pInterestingPids` coming from?

If that is the case, it might be the reason we end-up keeping the else part (and adding the GUC etc) ?

Reply:
All sessions used by the isolation test framework only connect to a single node. `pInterestingPids` are the pids of those sessions.